docs/api/shallow.md

```diff
@@ -47,9 +47,12 @@ describe('<MyComponent />', () => {
 1. `node` (`ReactElement`): The node to render
 2. `options` (`Object` [optional]):
   - `options.context`: (`Object` [optional]): Context to be passed into the component
-  - `options.disableLifecycleMethods`: (`Boolean` [optional]): If set to true, `componentDidMount`
-is not called on the component, and `componentDidUpdate` is not called after
-[`setProps`](ShallowWrapper/setProps.md) and [`setContext`](ShallowWrapper/setContext.md).
+  - `options.disableLifecycleMethods`: (`Boolean` [optional]): If set to true, most of the React
+lifecycle methods will never be executed: `componentDidMount`, `componentWillReceiveProps`,
+`shouldComponentUpdate`, `componentWillUpdate` and `componentDidUpdate`. Default to `false`.
+  - `options.lifecycleExperimental`: (`Boolean` [optional]): If set to true, the entire lifecycle
+(`componentDidMount` and `componentDidUpdate`) of the React component is called. The current default value
```

Comment:
@serut That's incorrect. Please refer to the test cases for `disableLifecycleMethods` https://github.com/airbnb/enzyme/blob/master/test/ShallowWrapper-spec.jsx#L2722
As tested there, if `disableLifecycleMethods` is `true` most lifecycle methods are called, `componentDidUpdate` and `componentDidMount` are the only two that are not called.

@ljharb this PR cannot be merged until the docs specify the correct list of lifecycle methods affected by either option.